packages/contractkit/src/address-registry.ts

```diff
@@ -0,0 +1,37 @@
+import { Address, AllContracts, CeloContract, NULL_ADDRESS } from 'src/base'
+import { newRegistry } from 'src/generated/Registry'
+import { Registry } from 'src/generated/types/Registry'
+import { ContractKit } from 'src/kit'
+
+// Registry contract is always predeployed to this address
+const REGISTRY_CONTRACT_ADDRESS = '0x000000000000000000000000000000000000ce10'
+
+export class AddressRegistry {
+  private readonly registry: Registry
+  private readonly cache: Map<CeloContract, Address> = new Map()
```

Comment:
Probably not something for v0, but what happens when this cache becomes outdated?

Reply:
can this happen? I mean, proxies won't change

Reply:
It's possible, registry pointers can change. Again, not something we need to address for now

Reply:
Ok
Added and issue for this https://github.com/celo-org/celo-monorepo/issues/575